src/main/java/com/nongfenqi/nexus/plugin/rundeck/RundeckMavenResource.java

```diff
@@ -77,79 +78,74 @@ public Response content(
             @QueryParam("r") String repositoryName,
             @QueryParam("g") String groupId,
             @QueryParam("a") String artifactId,
-            @QueryParam("v") String version,
-            @QueryParam("c") String classifier,
-            @QueryParam("p") @DefaultValue("jar") String extension
+            @QueryParam("v") String version
     ) {
-
-
-        // default version
-        version = Optional.ofNullable(version).orElse(latestVersion(
-                repositoryName, groupId, artifactId, classifier, extension
-        ));
-
-        // valid params
         if (isBlank(repositoryName) || isBlank(groupId) || isBlank(artifactId) || isBlank(version)) {
             return NOT_FOUND;
         }
 
-        Repository repository = repositoryManager.get(repositoryName);
-        if (null == repository || !repository.getFormat().getValue().equals("maven2")) {
-            return NOT_FOUND;
-        }
-
-        StorageFacet facet = repository.facet(StorageFacet.class);
-        Supplier<StorageTx> storageTxSupplier = facet.txSupplier();
-
-        log.debug("rundeck download repository: {}", repository);
-        final StorageTx tx = storageTxSupplier.get();
-        tx.begin();
-        Bucket bucket = tx.findBucket(repository);
-        log.debug("rundeck download bucket: {}", bucket);
-
-        if (null == bucket) {
-            return commitAndReturn(NOT_FOUND, tx);
+        List<Repository> repositories = getRepositories(repositoryName);
+
+        for (Repository repository : repositories) {
+
+            StorageFacet facet = repository.facet(StorageFacet.class);
+            Supplier<StorageTx> storageTxSupplier = facet.txSupplier();
+
+            log.debug("rundeck download repository: {}", repository);
+            final StorageTx tx = storageTxSupplier.get();
+            tx.begin();
+            Bucket bucket = tx.findBucket(repository);
+            log.debug("rundeck download bucket: {}", bucket);
+
+            if (null == bucket) {
```

Comment:
I don't understand what you are asking, please be more explicit. 
I also did not look at the code, I just copy pasted and build it. I assumed you checked. 

p.s. Downloading content does not work.

Reply:
😂 Sorry
I just paste a sample code yesterday.
The key code is 
```
        if (repository.getType() instanceof GroupType) {
            GroupFacet facet = repository.facet(GroupFacet.class);
            return facet.leafMembers();
        }
```
You need test it.

Reply:
ok. I will test it. Is there an easier way to doploy a new version? 
I tried putting the plugin in deploy folder but it did not pick it up. 
The manual way of doing install means that I have to copy files and change some test files. It should be easier than this with karaf.

Reply:
I've added some tests for version part. Download still doesn't work but it's not really needed since it's easy to build the url from GAV and download with curl:
````
curl -v -u user_name:password https://nexus.example.scom/repository/my-repo-name/group-id/artifact-id/3.7.3/artifact-id-3.7.3.tar.gz --output artifact-id.tar.gz

````